integration-commons/src/main/java/org/symphonyoss/integration/model/yaml/IntegrationProperties.java

```diff
@@ -269,6 +271,12 @@ public String getAgentUrl() {
     return String.format("https://%s:%s/agent", agent.getHost(), port);
   }
 
+  public String getApplicationUrl(String appId) {
+    Application application = getApplications().get(appId);
+
+    return String.format("https://%s/%s/%s", getIntegrationBridge().getHost(), APPS_CONTEXT, application.getContext());
```

Comment:
You should do a null-check to avoid NullPointerException when the configuration file doesn't have the IBridge host or application. We might return StringUtils.EMPTY in this scenario.